File: problem14/problem14.c
#include <stdio.h>

int main() {
	long int n=0;
	int current_chain, longest_chain=0, max_chain_number;

	for(int i=1; i<1000000; i++) {
		n = i;
		current_chain = 1;
		
		while(n != 1) {
			if(n % 2 == 0)
				n /= 2;
			else
				n = 3*n + 1;
			current_chain++;
		}

		if(current_chain > longest_chain) {
			longest_chain = current_chain;
			max_chain_number = i;
		}
	}
	printf("Result: %li\n", max_chain_number);
	return 0;
}

File: problem14/problem14.py
def chain(n):
	chain_lenght = 1
	while n != 1:
		if not n % 2:
			n /= 2
		else:
			n = n*3 + 1
		chain_lenght += 1
	return chain_lenght

if __name__ == '__main__':
	max_chain_lenght = 0
	for i in range(1, 1000000):
		current_chain = chain(i)
		if current_chain > max_chain_lenght:
			max_chain_lenght = current_chain
			max_number_chain = i
	print('Result:', max_number_chain)
